R/init.R

```diff
@@ -114,9 +114,11 @@ init <- function(data,
                  footer = tags$p(),
                  id = character(0)) {
   logger::log_trace("init initializing teal app with: data ({ class(data)[1] }).")
-  data <- teal.data::to_relational_data(data = data)
 
-  checkmate::assert_class(data, "TealData")
+  if (!inherits(data, c("TealData", "teal_data"))) {
+    data <- teal.data::to_relational_data(data = data)
+  }
```

Comment:
you can delegate this if into the method -> https://github.com/insightsengineering/teal.data/pull/178#discussion_r1365312526

I expect that you need to do this in many places so this could be more DRY

Reply:
let's continue discussing in the aforementioned thread - closing this one